tests/basic_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of the nmpolicy project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+
+	"github.com/nmstate/nmpolicy/nmpolicy"
+)
+
+func TestBasicPolicy(t *testing.T) {
+	t.Run("Basic policy", func(t *testing.T) {
+		testEmptyPolicy(t)
+		testPolicyWithOnlyDesiredState(t)
```

Comment:
Please add a test (with desired state) invoking `nmpolicy.GenerateState` twice.

Reply:
Am not clear what it is suppose to test exactly. The library only works on a given set of inputs, it has no affect if it is triggered more than once on the same input.